py/mpprint.h

```diff
@@ -80,3 +80,59 @@ int mp_vprintf(const mp_print_t *print, const char *fmt, va_list args);
 #endif
 
 #endif // MICROPY_INCLUDED_PY_MPPRINT_H
+
+// You can use several MP_PRN_LEVEL defines in separate _.c files
+#if defined(MP_PRN_LEVEL) && (MP_PRN_LEVEL > 0)
+// Debug messages during code developing with MP_PRN(level, ...) & MP_PRN_LEVEL.
+// An approximate hierarchy of debug levels MP_PRN_LEVEL is:
+#define MP_PRN_SUPPRESS 0 // SUPPRESS all messages. Use it in the release version.
+#define MP_PRN_CRITICAL 1 // For the most CRITICAL errors, often requiring a system reset. Use a message with this level, if possible, raising an exception.
+#define MP_PRN_ERROR 2 // ERROR requiring program restart, use message with this level before raising an exception.
+#define MP_PRN_WARNING 3 // WARNING, something went wrong, but you can fix it with additional operations in code right now or may ignore it.
+#define MP_PRN_INFO 4 // INFO, it is interesting and useful for understanding a bug.
+#define MP_PRN_DEBUG 5 // DEBUG, more detailed information, dig deeper.
+#define MP_PRN_TRACE 6 // TRACE, show a flow of the algorithm, like enter/exit a function.
+// In reality, you may use your own classification of debug levels.
+
+#if defined(MP_PRN)
+#undef MP_PRN
```

Comment:
I think this under should be removed, and instead define the command below only if it's not already defined. It should ideally be possible for people to define their own command in mpconfigboard.h

Reply:
`mprint.h` was included early in files and `MP_DEBUG_PRINT` was defined previously as nothing
```
#define MP_DEBUG_PRINT(level, ...) 
```
so we need undef and then define MP_DEBUG_PRINT again
```
#if defined(MP_DEBUG_PRINT)
#undef MP_DEBUG_PRINT
#endif

#define MP_DEBUG_PRINT(level, ...) \
    do { \
     ...
```


Unfortunately, this blocks redefining in mpconfigboard.h